aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -1902,6 +1920,120 @@ In the following `App` component example:
 > [!NOTE]
 > Not throwing if the cancellation token in <xref:Microsoft.AspNetCore.Components.Routing.NavigationContext> is canceled can result in unintended behavior, such as rendering a component from a previous navigation.
 
+## Handle location changes in a component
+
+`RegisterLocationChangingHandler` registers a handler to process incoming navigation events. The handler's context provided by `LocationChangingContext` includes the following properties:
+
+* `TargetLocation`: Gets the target location.
+* `HistoryEntryState`: Gets the state associated with the target history entry.
+* `IsNavigationIntercepted`: Gets whether the navigation was intercepted from a link.
+* `CancellationToken`: Gets a <xref:System.Threading.CancellationToken> to determine if the navigation was canceled, for example, to determine if the user triggered a different navigation.
+* `PreventNavigation`: Called to prevent the navigation from continuing.
+
+A component can register multiple location changing handlers. Navigations invoke all of the location changing handlers registered across the entire app (across multiple components), and any internal navigation executes them all in parallel. In addition to <xref:Microsoft.AspNetCore.Components.NavigationManager.NavigateTo%2A> invoking the handlers, selecting internal links (those that point to URLs under the app's base path) and navigating using the forward/back buttons also invoke the handlers.
```

Comment:
```suggestion
A component can register multiple location changing handlers. Navigations invoke all of the location changing handlers registered across the entire app (across multiple components), and any internal navigation executes them all in parallel. In addition to <xref:Microsoft.AspNetCore.Components.NavigationManager.NavigateTo%2A> invoking the handlers, selecting internal links (those that point to URLs under the app's base path) and navigating using the forward/back buttons also invokes the handlers.
```

Reply:
Plural verb for compound subject joined by 'and' ... *selecting internal links* **_AND_** navigating ... invoke.

However, I think this is a bit long for us (i.e., to control loc of the content). I think I'll use a list and break this up.